authentication/native.go

```diff
@@ -0,0 +1,109 @@
+package authentication
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/blockchain"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/workflows"
+)
+
+// ArgsNativeAuthClient -
+type ArgsNativeAuthClient struct {
+	TxSigner             blockchain.TxSigner
+	ExtraInfo            interface{}
+	Proxy                workflows.ProxyHandler
+	PrivateKey           crypto.PrivateKey
+	TokenExpiryInSeconds uint64
+	Host                 string
+}
+
+type nativeAuthClient struct {
+	txSigner             blockchain.TxSigner
+	encodedExtraInfo     string
+	proxy                workflows.ProxyHandler
+	skBytes              []byte
+	tokenExpiryInSeconds uint64
+	encodedAddress       string
+	encodedHost          string
+	token                string
+	tokenExpire          time.Time
+}
+
+// NewNativeAuthClient will create a new native client able to create authentication tokens
+func NewNativeAuthClient(args ArgsNativeAuthClient) (AuthClient, error) {
+	if check.IfNil(args.TxSigner) {
+		return nil, ErrNilTxSigner
+	}
+
+	publicKey := args.PrivateKey.GeneratePublic()
+	publicKeyBytes, err := publicKey.ToByteArray()
+	if err != nil {
+		return nil, err
+	}
+
+	address := data.NewAddressFromBytes(publicKeyBytes)
+
+	extraInfoBytes, err := json.Marshal(args.ExtraInfo)
+	if err != nil {
+		return nil, err
+	}
+	encodedExtraInfo := base64.StdEncoding.EncodeToString(extraInfoBytes)
+
+	encodedAddress := base64.StdEncoding.EncodeToString(address.AddressBytes())
+	skBytes, err := args.PrivateKey.ToByteArray()
```

Comment:
From linter: ineffectual assignment to err (ineffassign)

Reply:
done